src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -287,131 +284,59 @@ public static IResourceBuilder<AzureServiceBusResource> RunAsEmulator(this IReso
-                        foreach (var subscription in topic.Subscriptions)
-                        {
-                            writer.WriteStartObject();              //               "{ (Subscription)"
-                            subscription.WriteJsonObjectProperties(writer);
-
-                            writer.WriteStartArray("Rules");        //                 "Rules": [
-                            foreach (var rule in subscription.Rules)
-                            {
-                                writer.WriteStartObject();
-                                rule.WriteJsonObjectProperties(writer);
-                                writer.WriteEndObject();
-                            }
-
-                            writer.WriteEndArray();                 //                  ] (/Rules)
-
-                            writer.WriteEndObject();                //               } (/Subscription)
-                        }
-
-                        writer.WriteEndArray();                     //             ] (/Subscriptions)
-
-                        writer.WriteEndObject();                    //           } (/Topic)
-                    }
-                    writer.WriteEndArray();                         //         ] (/Topics)
+            var hasCustomConfigJson = builder.Resource.Annotations.OfType<ContainerMountAnnotation>().Any(v => v.Target == AzureServiceBusEmulatorResource.EmulatorConfigJsonPath);
 
-                    writer.WriteEndObject();                        //       } (/Namespace)
-                    writer.WriteEndArray();                         //     ], (/Namespaces)
-                    writer.WriteStartObject("Logging");             //     "Logging": {
-                    writer.WriteString("Type", "File");             //       "Type": "File"
-                    writer.WriteEndObject();                        //     } (/LoggingConfig)
+            if (hasCustomConfigJson)
+            {
+                return Task.CompletedTask;
+            }
 
-                    writer.WriteEndObject();                        //   } (/UserConfig)
-                    writer.WriteEndObject();                        // } (/Root)
-                }
+            // Create Config.json file content and its alterations in a temporary file
+            var tempConfigFile = WriteEmulatorConfigJson(builder.Resource);
 
+            try
+            {
                 // Apply ConfigJsonAnnotation modifications
-                var configJsonAnnotations = emulatorResource.Annotations.OfType<ConfigJsonAnnotation>();
+                var configJsonAnnotations = builder.Resource.Annotations.OfType<ConfigJsonAnnotation>();
 
                 foreach (var annotation in configJsonAnnotations)
                 {
-                    using var readStream = new FileStream(configFileMount.Source!, FileMode.Open, FileAccess.Read);
+                    using var readStream = new FileStream(tempConfigFile, FileMode.Open, FileAccess.Read);
```

Comment:
Do we need to read and write the file for each configJsonAnnotation? Why not read it once, run each annotation, and then write it back once?